dids/resolver.go

```diff
@@ -0,0 +1,41 @@
+package dids
+
+import "sync"
+
+// Resolve resolves the provided DID URI. This function is capable of resolving
+// the DID methods implemented in web5-go
+func Resolve(uri string) ResolutionResult {
```

Comment:
I read your explanation in the PR, but it's really non-idiomatic to not return an error. For Go, I think what I would do is return `(ResolutionResult, error)`, and make `ResolutionResult` implement `error`. An alternative  and probably preferable solution would be to have another type `ResolutionError` that has all of the correct fields for the protocol, but also implements `error`.

Reply:
@aat went with the 2nd suggestion: https://github.com/TBD54566975/web5-go/commit/2bf2743e6d43dce1ae12f2cfb6c3a58d2fe87e9d

> An alternative and probably preferable solution would be to have another type ResolutionError that has all of the correct fields for the protocol, but also implements error

lmk if the change doesn't reflect what you had in mind

Reply:
Perfect!